src/Provider/DecoratingRedisSettingsProvider.php

```diff
@@ -244,4 +247,36 @@ private function buildHashmap(bool $force = false, ?string $domainName = null)
             }
         }
     }
+
+    /**
+     * @return DomainModel[]
+     */
+    private function getPersistedDomains(): array
+    {
+        $redisKeys = array_flip($this->redis->keys(sprintf('%s\[*\]', $this->namespace)));
```

Comment:
A number of sources, including the official Redis documentation, note that using the KEYS command is a bad idea in production environments due to possible blocking.